projects/RabbitMQ.Client/client/impl/ConsumerDispatching/ConsumerDispatcherBase.cs

```diff
@@ -8,7 +10,7 @@ namespace RabbitMQ.Client.ConsumerDispatching
     internal abstract class ConsumerDispatcherBase
     {
         private static readonly FallbackConsumer s_fallbackConsumer = new FallbackConsumer();
-        private readonly Dictionary<string, IBasicConsumer> _consumers = new Dictionary<string, IBasicConsumer>();
+        private readonly IDictionary<string, IBasicConsumer> _consumers = new ConcurrentDictionary<string, IBasicConsumer>();
```

Comment:
Beware that this constructor of `ConcurrentDictionary<TKey, TValue>` has a concurrency level of the number of number of "processors" ([`Environment.ProcessorCount` property](https://learn.microsoft.com/en-us/dotnet/api/system.environment.processorcount)), and that determines the size of lock array.